client/app/states/dashboard/dashboard.state.js

```diff
@@ -90,34 +130,38 @@
     if (!$state.navFeatures.services.show) {
       return undefined;
     }
-    var options = {expand: false, filter: ['retired=true'] };
+    var options = {expand: false, filter: ['service_id=nil', 'retired=true'] };
 
     return CollectionsApi.query('services', options);
   }
 
   /** @ngInject */
-  function resolveNonRetiredServices(CollectionsApi, $state) {
+  function resolveServicesWithDefinedServiceIds(CollectionsApi, $state) {
     if (!$state.navFeatures.services.show) {
       return undefined;
     }
-    var options = {expand: false, filter: ['retired=false'] };
+    var options = {expand: false, filter: ['service_id=nil'] };
 
     return CollectionsApi.query('services', options);
   }
 
-  /** @ngInject */
-  function resolveServicesWithDefinedServiceIds(CollectionsApi, $state) {
-    if (!$state.navFeatures.services.show) {
-      return undefined;
+  function chainRequestPromises(promiseArray, vm, type) {
```

Comment:
Thanks @himdel -- `$q` was an elegant approach alright!
I also liked the idea of using `sumBy`...however, had to bump up the version of `lodash` to 4.12.0 to be able to use that.